File: 1566.detect-pattern-of-length-m-repeated-k-or-more-times.py
#
# @lc app=leetcode id=1566 lang=python
#
# [1566] Detect Pattern of Length M Repeated K or More Times
#

# @lc code=start
class Solution(object):
    def containsPattern(self, arr, m, k):
        """
        :type arr: List[int]
        :type m: int
        :type k: int
        :rtype: bool
        """

        pattern = ""
        for i in range(len(arr)):
            current_character = arr[i]
            for j in range(1, len(arr)):
                if i == len(arr[j]) or current_character != arr[j][i]:
                    return pattern

            pattern += current_character

        return pattern

# @lc code=end

File: 1566.detect-pattern-of-length-m-repeated-k-or-more-times.js
/*
 * @lc app=leetcode id=1566 lang=javascript
 *
 * [1566] Detect Pattern of Length M Repeated K or More Times
 */

// @lc code=start
/**
 * @param {number[]} arr
 * @param {number} m
 * @param {number} k
 * @return {boolean}
 */
var containsPattern = function(arr, m, k) {
    const n = arr.length;
    for (let i = 0; i <= n - m * k; i++) {
        let flag = true;
        for (let j = 0; j < m * k; j++) {
            if (arr[i + j] !== arr[i + j % m]) {
                flag = false;
                break;
            }
        }
        if (flag) return true;
    }
    return false;
};
// @lc code=end
